Hds 1794 header docs

harriplappalainen opened the pull request:
## Description

- Document Header components as they are now
  - Extensive docs for sub components props as they can't be seen in storybook
  - NavigationSearch and NavigationLanguageSelector components are still up in the air on how they're going to end up
- Small fixes to Header components JSDocs
- Change prop names to href instead of url. It's semantically correct and is more consistent with HDS
- Add prop tables for Footer components as well
- Small fixes to Footer components JSDocs

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-1794

## How Has This Been Tested?
Locally, running tests
[Demo](https://city-of-helsinki.github.io/hds-demo/header-docs)

To do:
- [x] Docs about ActionBarItem

laurakarhu commented on site/src/docs/components/header/index.mdx:
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
 
-#### With navigation bar
+#### With universal bar
 
```
Universal bar is aimed for frequenlty visited and useful links, e.g. feedback and news.

laurakarhu commented on site/src/docs/components/header/index.mdx:
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
+</PlaygroundPreview>
 
 #### With navigation language selector
 
```
Language selector changes the language for the whole service. It is recommended to offer full Finnish, Swedish and English versions. Other languages can be added based on the need and target audience of the service. Smaller subset of content can be separated from fully translated content.

laurakarhu commented on site/src/docs/components/header/index.mdx:
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
+- On smaller screens (768px and down) the Header.NavigationLanguageSelector moves under the Header.ActionBar.
 
 #### With skip link
 
```
Use the skip link component to help keyboard-only users skip to the main content on a page.

laurakarhu commented on site/src/docs/components/header/index.mdx:
```diff
@@ -22,56 +22,232 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
-- See example for Header.SkipLink <Link openInNewTab openInNewTabAriaLabel="Opens in a new tab" href="/storybook/react/?path=/story/components-header--with-skip-link">in Storybook</Link>
-- The Header.SkipLink supports ariaLabel which gives possibility to define more informative text for screen readers
+- The Header.SkipLink supports `ariaLabel` prop which gives possibility to define more informative text for screen readers.
+
```
It is recommneded to have the skip link, especially if there is any other component in addition to the action bar.

laurakarhu reviewed (commented):
Good job, I had couple of suggestions to enhance the documentation.

Also, still thinking of the order:should it be the same as in the header or should it be mandatory first as it is now?

harriplappalainen commented:
@laurakarhu Thank you for comments! I fixed them and added some more good stuff. Check it out.
Additional fixes:
- Old Navigation component has link to new Header docs
- Added StatusLabelTooltip
- Info about ActionBar's `titleStyle` prop usage
- Info about UniversalBar's primaryLink usage

laurakarhu commented on site/src/docs/components/header/index.mdx:
```diff
@@ -17,61 +17,289 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
+- Try to keep the order of actions (language select, search, log in) consistent in navigation. Do not change the default order without a good reason.
 
 ### Variations
 
```
Should we mention here also that the order to make it clear?
Maybe also a list with anchor links


Variations are presented in the same order than they appear in the actual header.
- With skip link
- With universal bar
- With action bar
- With navigation bar
- With navigation language selector
- With search

laurakarhu commented on site/src/docs/components/header/index.mdx:
```diff
@@ -17,61 +17,289 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
-- HeaderActionBar is coming soon
 - The inner components should **always** be in this order: Header.SkipLink, Header.UniversalBar, HeaderActionBar and HeaderNavigationBar.
+- Try to keep the order of actions (language select, search, log in) consistent in navigation. Do not change the default order without a good reason.
 
```
To better understand how navigation elements work and how they should be combined, see also Navigation pattern (link)

laurakarhu commented on site/src/docs/components/header/accessibility.mdx:
```diff
@@ -16,3 +16,4 @@ export default ({ children, pageContext }) => <TabsLayout pageContext={pageConte
 - HDS Navigation is designed to support many different brand colours. When customising colours, refer to <InternalLink href="/foundation/design-tokens/colour">colour guidelines</InternalLink> to ensure accessibility.
 - **When designing a menu link hierarchy, always think about keyboard and screen reader users.** First, make sure top-level menu labels are clear, and submenu items contextually fit under it. Next, make sure the menu option order is logical and reasonable. Remember that keyboard users need to manually navigate to each element and thus placing a commonly used menu option last is not optimal.
 - Landmarks are provided by the wrapping `<header>` element, `<nav>` element in the Header.NavigationMenu component and with `role="search"` attribute on the Header.NavigationSearch component. In case you want more landmarks, the Header.UniversalBar, Header.ActionBar and Header.NavigationSearch components support `role` prop which should be used together with a descriptive `ariaLabel` prop. Just  ...
+- Header.NavigationMenu's dropdowns open only by clicking them. They don't open from hovering with the mouse as it can be difficult to navigate for people with motor and seeing challenges.
```
Even though HDS header supports larger needs, only use the elements you need. Avoid clutter.

laurakarhu reviewed (commented):
Looking really good, just couple of suggestions to add still, links and notes.

laurakarhu reviewed (approved):
Hyvää pyysin, mutta priimaa pukkaa! Kiitos!

mrTuomoK reviewed (approved):
I don't see anything suspicious here 😄 a Goood good 👍
